src/Corax/Querying/Matches/SortingMatches/SortingMultiMatch.Comparers.cs

```diff
@@ -96,6 +119,8 @@ public void Init(ref SortingMultiMatch<TInner> match, UnmanagedSpan<long> batchR
             where TComparer2 : struct, IComparer<UnmanagedSpan>, IComparer<int>, IEntryComparer
             where TComparer3 : struct, IComparer<UnmanagedSpan>, IComparer<int>, IEntryComparer
         {
+            Debug.Assert(_comparerId == 0, "_comparerId == 0");
```

Comment:
Can you explain why you are forced to be the first comparer here? What validates this?

Reply:
SortBatch is made to read terms from disk (in case of score to calculate them). It's called only once in the codebase, on TComparer1 instance. So I added this assertion.